moped-api/users/validation.py

```diff
@@ -78,7 +78,7 @@
         "required": True,
         "minlength": 8,
         "maxlength": 32,
-        "regex": "^[a-zA-Z0-9_\-\!\@\%\^\*\~\?\.\:\&\*\(\)\[\]\$]*$",
+        "regex": "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$",
```

Comment:
One thing that I wanted to ask about -- do these [password requirements](https://github.com/cityofaustin/atd-moped/blob/91f87e25e4bd155522a412cdce89e006f7f18753/moped-editor/src/views/staff/helpers.js#L57) need to be reworded with respect to the improved regex?

Reply:
I need to test how the errors come back, because I am not sure the password requirements error is actually showing up -- unless cognito returns error messages like `value does not match regex '^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+$'"`

Reply:
These are great questions, and I think that you are right, @frankhereford, and that error message needs to be updated. I have PR in the works with UI refactor and I can sneak that in. This might even be worth its own issue since there is separate password regex [here](https://github.com/cityofaustin/atd-moped/blob/91f87e25e4bd155522a412cdce89e006f7f18753/moped-editor/src/views/staff/StaffForm.js#L232-L235) and a different error message set [here](https://github.com/cityofaustin/atd-moped/blob/91f87e25e4bd155522a412cdce89e006f7f18753/moped-editor/src/views/staff/StaffForm.js#L255-L259) that are aimed at user reactivation. The password requirements are the same everywhere so they should all be aligned. Thank y'all!

Reply:
@chiaberry I also think that you are right about the error messaging being broken. That also needs to be looked at. 🙏

Reply:
I forgot to add that I created https://github.com/cityofaustin/atd-data-tech/issues/10085 to come back and check that the error messaging is displaying as we would expect.